src/object.c

```diff
@@ -692,28 +692,35 @@ char *strEncoding(int encoding) {
 
 /* =========================== Memory introspection ========================== */
 
+static size_t sizeOfStringObject(robj *o) {
+    size_t strSize = 0;
+    if (o->encoding == OBJ_ENCODING_INT) {
+        strSize = sizeof(long long);
```

Comment:
If you see the implementation at https://github.com/antirez/redis/blob/unstable/src/object.c#L124-L139, long long value is stored in the o->ptr only. o->ptr does not refer to any other location. "value" is first casted to void pointer and stored in the o->ptr. I am not sure I am able to explain it well, but there is not other memory allocated except the robj. o->ptr itself holds the long long value.
o->ptr = (void*)((long)value)

You can also look into how the value is read back at https://github.com/antirez/redis/blob/unstable/src/object.c#L569